src/components/LinkableText.tsx

```diff
@@ -27,18 +42,33 @@ export const LinkableText: React.FC<Props> = ({text, ...props}) => {
 
   return (
     <Text style={props.style}>
-      {parts.map((part, index) =>
-        urlRegex.test(part?.toLocaleLowerCase()) ? (
-          <Text
-            key={index}
-            style={{color: COLORS.blueLink, textDecorationLine: 'underline'}}
-            onPress={() => handlePress(part)}>
-            {`${part} `}
-          </Text>
-        ) : (
-          <Text key={index}>{`${part} `}</Text>
-        )
-      )}
+      {parts.map((part, index) => {
+        if (urlRegex.test(part?.toLocaleLowerCase())) {
+          return (
+            <Text
+              key={index}
+              style={{color: COLORS.blueLink, textDecorationLine: 'underline'}}
+              onPress={() => handlePress(part)}>
+              {index === parts.length - 1 ? `${part}` : `${part} `}
+            </Text>
+          );
+        }
+        if (part.startsWith('#') && props.withTopicDetection) {
+          return (
+            <Text key={index}>
+              <TopicDetectionOnChat
+                text={part}
+                multipart={parts?.length > 1}
+                onPress={handleTopicPress}
+                isFirstLine={props?.isFirstLine && index === 0}
+              />
+              <Text> </Text>
+            </Text>
+          );
+        }
+
+        return <Text key={index}>{`${part} `}</Text>;
+      })}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic in this section seems complex and hard to follow due to nested conditionals and multiple returns. It would be beneficial to refactor this into smaller, more manageable functions. This will improve readability and maintainability of the code. Here's a suggestion on how you might approach this:

```diff
- {parts.map((part, index) => {
-   if (urlRegex.test(part?.toLocaleLowerCase())) {
-     return (
-       <Text
-         key={index}
-         style={{color: COLORS.blueLink, textDecorationLine: 'underline'}}
-         onPress={() => handlePress(part)}>
-         {index === parts.length - 1 ? `${part}` : `${part} `}
-       </Text>
-     );
-   }
-   if (part.startsWith('#') && props.withTopicDetection) {
-     return (
-       <Text key={index}>
-         <TopicDetectionOnChat
-           text={part}
-           multipart={parts?.length > 1}
-           onPress={handleTopicPress}
-           isFirstLine={props?.isFirstLine && index === 0}
-         />
-         <Text> </Text>
-       </Text>
-     );
-   }
- 
-   return <Text key={index}>{`${part} `}</Text>;
- })}

+ {parts.map((part, index) => renderPart(part, index))}

// Add these helper functions
function renderPart(part, index) {
  if (urlRegex.test(part?.toLocaleLowerCase())) {
    return renderLink(part, index);
  }
  if (part.startsWith('#') && props.withTopicDetection) {
    return renderTopic(part, index);
  }
  return renderText(part, index);
}

function renderLink(part, index) {
  return (
    <Text
      key={index}
      style={{color: COLORS.blueLink, textDecorationLine: 'underline'}}
      onPress={() => handlePress(part)}>
      {index === parts.length - 1 ? `${part}` : `${part} `}
    </Text>
  );
}

function renderTopic(part, index) {
  return (
    <Text key={index}>
      <TopicDetectionOnChat
        text={part}
        multipart={parts?.length > 1}
        onPress={handleTopicPress}
        isFirstLine={props?.isFirstLine && index === 0}
      />
      <Text> </Text>
    </Text>
  );
}

function renderText(part, index) {
  return <Text key={index}>{`${part} `}</Text>;
}
```
This way, the logic for rendering each part is separated into its own function, making it easier to understand and modify in the future.


<!-- This is an auto-generated comment by OSS CodeRabbit -->